src/components/AuthLayout/index.js

```diff
@@ -0,0 +1,162 @@
+import React, { Component, Fragment } from 'react';
+import {
+  Grid,
+  GridItem,
+  Title,
+  Flex,
+  FlexItem,
+  Dropdown,
+  DropdownToggle,
+  Button,
+} from '@patternfly/react-core';
+import { connect } from 'dva';
+import { CaretDownIcon } from '@patternfly/react-icons';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faAngleLeft } from '@fortawesome/free-solid-svg-icons';
+import { routerRedux } from 'dva/router';
+import styles from './index.less';
+import logo from '../../assets/white.svg';
+
+@connect(user => ({
+  user: user.user,
+}))
+class AuthLayout extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      isOpen: false,
+    };
+  }
+
+  onToggle = isOpen => {
+    this.setState({
+      isOpen,
+    });
+  };
+
+  onSelect = () => {
+    const { isOpen } = this.state;
+    this.setState({
+      isOpen: !isOpen,
+    });
+  };
+
+  navigateToHome = () => {
+    const { dispatch } = this.props;
+    dispatch(routerRedux.push(`/`));
+  };
+
+  render() {
+    const { toPreview, heading, backOpt } = this.props;
+    const { isOpen } = this.state;
+
+    const back = (
+      <Button
+        variant="link"
+        icon={<FontAwesomeIcon icon={faAngleLeft} />}
+        className={styles.inlineLink}
+        style={{ padding: '0 0 20px 5px' }}
+        onClick={() => this.navigateToHome()}
+      >
+        Back
+      </Button>
+    );
+    const Heading = (
+      <div>
+        <Title headingLevel="h2" size="3xl" className={styles.section}>
+          {backOpt === 'true' ? back : <Fragment />}
+          <br />
+          {heading}
+          <Dropdown
+            style={{ float: 'right' }}
+            onSelect={this.onSelect}
+            toggle={
+              <DropdownToggle
```

Comment:
I'm receiving the following error when opening this DropdownToggle: 
`Uncaught Error: Element type is invalid: expected a string (for built-in components) or a class/function (for composite components) but got: undefined. You likely forgot to export your component from the file it's defined in, or you might have mixed up default and named imports.`

Reply:
fixed